### ISSUE TITLE
Feature request: Run once, then exit

### ISSUE THREAD

**Issue by icopp** [Qard/onchange#118]
`onchange` is very close to being usable as a general-purpose file runner across glob matches in a way I haven't seen with other npm tools.

Here's a quick example of the kind of thing I'm looking at, using https://github.com/drwpow/openapi-typescript to use a schema at `docs/whatever.yml` to generate `src/generated/whatever.ts`:

```
  "dev:openapi": "onchange -a 'docs/*.yml' -- openapi-typescript {{file}} --prettier-config . -o src/generated/{{fileBaseNoExt}}.ts",
```

If there was a flag to only run that once (without actually watching for further changes), I could use the same functionality in build steps without needing to break out any other script stuff.

**Comment by blakeembrey** [comment]
This sounds interesting, could be done. We might be able to use this with chokidar by closing the watcher on `ready`, otherwise we could make a PR to `chokidar` or use a different library with the same underlying glob support. I think the flag is only useful in combination with `-a` in this case, does that sound correct?

**Comment by icopp** [comment]
For this I'd think a new self-contained flag, maybe `-r` for 'run' or something.

**Comment by Qard** [comment]
I'd prefer something a little more obvious like `--once`, but other than that I like the idea. :)

**Comment by forivall** [comment]
Yeah, i was just thinking about this feature too -- luckily `--outpipe` solved the need i had for it.

`onchange -v -o 'dot -Tpng | imgcat' *.dot -- cat '{{file}}'` (where `cat '{{file}}' is actually a different command that generates graphviz, but nbd)

The alternative solution using `once` would be
```sh
while changed=$(onchange --once *.dot) ; do < dot -Tpng | imgcat ; done
```

**Comment by blakeembrey** [comment]
@forivall Can you help me understand your example a little more? It sounds like a different request from the `while` loop but maybe I don’t understand. Also not clear to me how outpipe helped you exit on ready, which is what @icopp was trying to do - basically run a command with each file as a glob instead of a file watcher.
